### PR TITLE
refactor(menu-item,menu): keyboard interaction

### PR DESCRIPTION
**Pull request by anveshmekala**
**Related Issue:** #6531 

## Summary
Refactors keyboard navigation and cleanup

### CODE REVIEW

File: src/components/menu-item/menu-item.tsx [hunk 1]
```diff
@@ -172,8 +177,6 @@ export class CalciteMenuItem implements LoadableComponent {
     this.active = this.active || this.editable;
     this.isFocused = this.active;
     // todo just get any nav items in the default slot?
-    // this.hasSubMenu = this.hasSlottedItems();
-
     // for now to detect nesting only working two level for demo.need to just check if it has any parent originating at top lvel
```
Comment (macandcheese) [inline src/components/menu-item/menu-item.tsx:180]: Can we remove any todo / wip items (I think these are probably mine, ha)

File: src/components/menu-item/menu-item.tsx [hunk 2]
```diff
@@ -504,9 +398,8 @@ export class CalciteMenuItem implements LoadableComponent {
               : "chevron-down"
             : dirChevron
         }
-        onClick={() => (this.open = !this.open)}
+        onClick={this.clickHandler}
```
Comment (macandcheese) [inline src/components/menu-item/menu-item.tsx:401]: 👍

**Review by macandcheese — APPROVED** [review]
Excellent!

File: src/components/menu-item/menu-item.tsx (hunk 1 above)
Comment (anveshmekala) [reply src/components/menu-item/menu-item.tsx:180]: i left them to keep a track of whats leftover. Are these irrelevant?

Comment (anveshmekala) [reply src/components/menu-item/menu-item.tsx]: removed them.

**Pull request merged**